include/net/wifi_defs.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef WIFI_DEFS_H_
+#define WIFI_DEFS_H_
+
+/** @file wifi_defs.h */
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/wifi.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup wifi_defs common Wi-Fi components
+ * @{
+ */
+
+/* 2 chars per byte, colon separated */
+#define WIFI_MAC_ADDR_STR_LEN ((WIFI_MAC_ADDR_LEN * 2) + 5)
+
+struct wifi_ap_info {
+	char mac_addr_str[WIFI_MAC_ADDR_STR_LEN + 1];
+	char ssid_str[WIFI_SSID_MAX_LEN + 1];
+	uint8_t channel;
+	int8_t rssi;
+};
+
+struct wifi_scan_info {
+	struct wifi_ap_info *ap_info;
+	uint16_t cnt;
+};
```

Comment:
Location library could import them from within an `nrf_cloud` path, right? I just don't think they should be declared/named as root-level structs, since they compete with existing Wi-Fi structs at that level

Reply:
Maybe another solution would be to have them declared under the location lib (`wifi_location_defs.h`?) and have nrf_cloud import those defs specifically

Reply:
i will defer to @jhirsi 
i have no preference

Reply:
I have no strong preference either for this. This was just one possible way to do it to align the code that Justin did in a past when I demoed this (WiFi positioning over MQTT). Yes, maybe wifi_scan_result could be also used and wrap those with the information needed in addition to wifi_scan_result.
 
Target was to have data structures containing wifi positioning data as cloud agnostic within ncs and it would be convenient that all wifi cloud services would use that, including nrf cloud. The other wifi service specific codes are in location lib and also possible in application level. But yes, common factor for both kind of usages is a location lib.

Reply:
ok, updated to use `wifi_scan_result`.
`wifi_defs.h` has been removed.
`wifi_scan_info` is now in `nrf_cloud_location.h`
@jhirsi please review, thanks!

Reply:
Apologies, I should have been clearer in this in my prev comment, but I'd like to use also wifi_scan_info with other than nrf cloud also, and thus it is not convenient that it is in nrf_cloud named header.
So, maybe I have a little bit stronger preference for this after all and it should be global to have commons for all users. Maybe name could be wifi_location_common.h or the original, for that I do not have strong preference.

Reply:
I'm wondering if the solution here might be to have two definitions of some kind of a struct which represents "sets of wifi_scan_info", one for use with nrf_cloud functions, and one for use with the location library, and have the location library translate between its definition, and the nrf_cloud definition when using the nrf_cloud integration.

At least, that is how this problem would be handled if we treat location and nrf_cloud as totally separate libraries. Trying to re-use the same struct is breaking our compartmentalization and leading to stranded definition files in the global scope

Reply:
updated.